dsgrid/config/dataset_config.py

```diff
@@ -424,32 +401,50 @@ def model_class():
         return DatasetConfigModel
 
     @classmethod
-    def load(cls, config_file, dimension_manager):
+    def load_from_registry(cls, config_file, dimension_manager, registry_path):
```

Comment:
NIT: `registry_data_path` instead of `registry_path`? When I see registry_path, I think of `.dsgrid-registry` but it's really one level down.